src/server/api/routes/users/v2/follow.js

```diff
@@ -9,41 +9,61 @@ const {
   nftMintSubscriptionCollection,
 } = require('../../../../modules/firebase');
 const { publisher, PUBSUB_TOPIC_MISC } = require('../../../../modules/pubsub');
+const { getUserFollowees } = require('../../../util/api');
 
 const router = Router();
 
 router.get('/followees', authenticateV2Login, async (req, res, next) => {
   try {
     const { user } = req.session;
-    const userDoc = await walletUserCollection.doc(user).get();
-    if (!userDoc.exists) {
-      res.json({ followees: [] });
-      return;
-    }
-    const {
-      followees: walletFollowees = [],
-      email,
-      pastFollowees = [],
-    } = userDoc.data();
-    let legacyFollowees = [];
-    if (email) {
-      const snapshot = await nftMintSubscriptionCollection
-        .where('subscriberEmail', '==', email)
-        .get();
-      legacyFollowees = snapshot.docs.map(doc => {
-        const { subscribedWallet } = doc.data();
-        return subscribedWallet;
-      });
-    }
-    const followees = [
-      ...new Set([...walletFollowees, ...legacyFollowees]).values(),
-    ];
+    const { followees, pastFollowees } = await getUserFollowees(user);
     res.json({ followees, pastFollowees });
   } catch (err) {
     handleRestfulError(req, res, next, err);
   }
 });
 
+router.get(
+  '/suggested-followees',
+  authenticateV2Login,
+  async (req, res, next) => {
+    try {
+      const { user } = req.session;
+      const { initialList = [] } = req.body;
+      const { followees = [], pastFollowees = [] } = await getUserFollowees(
+        user
+      );
+
+      const currentFollowees = [...followees, ...pastFollowees];
+      const currentFolloweesSet = new Set(currentFollowees);
+
+      const suggestedSet = new Set(initialList);
+      let currentIndex = 0;
+
+      // If we don't have enough suggestions, fetch more
+      while (suggestedSet.size < 3 && currentIndex < currentFollowees.length) {
+        // eslint-disable-next-line no-await-in-loop
+        const { followees = [], pastFollowees = [] } = await getUserFollowees(
+          currentFollowees[currentIndex]
+        );
+
+        const followSuggestions = [...followees, ...pastFollowees].filter(
+          f => !currentFolloweesSet.has(f)
+        );
+
+        followSuggestions.forEach(suggestion => suggestedSet.add(suggestion));
+        currentIndex += 1;
```

Comment:
We would want avoid just leaking the whole follower list of `currentIndex` to user. Thus we should limit and perform some randomization of the output

- Size of `currentFollowees` must be larger than a number, say 3, otherwise following 1 user would always yield result from that particular list
- `currentFollowees` should contain some followees that are not already in `currentFolloweesSet` in order to be counted towards to the size limit above
- We should randomly draw from `currentFollowees` instead of using `currentIndex`

If the above conditions are not met, we should just return default suggestions/nothing.

Also note these rules are still not perfect, a better idea would be to use public information for suggestions, instead of non-public information like follower list i.e. use creators of NFT owned by a user